web/src/HUD.tsx

```diff
@@ -107,11 +104,17 @@ class HUD extends Component<HudProps, HudState> {
   }
 
   componentDidMount() {
-    this.controller.createNewSocket()
+    if (this.pathBuilder.isSnapshot()) {
+      this.controller.setStateFromSnapshot()
+    } else {
+      this.controller.createNewSocket()
+    }
   }
 
   componentWillUnmount() {
-    this.controller.dispose()
+    if (!this.pathBuilder.isSnapshot()) {
+      this.controller.dispose()
```

Comment:
i don't understand why the controller is conditionally disposed. seems like it should always be disposed?

Reply:
Well it doesn't do anything in non-websocket contexts, but also won't crash if the socket isn't set so I'll remove the conditional here as it is unnecessary.